ompi/request/request.h

```diff
@@ -136,6 +137,16 @@ struct ompi_request_t {
     ompi_request_complete_fn_t req_complete_cb; /**< Called when the request is MPI completed */
     void *req_complete_cb_data;
     ompi_mpi_object_t req_mpi_object;           /**< Pointer to MPI object that created this request */
+#if 1 || OPAL_ENABLE_FT_MPI
```

Comment:
??? and clean the comment below

Reply:
This would make ENABLE_FT_MPI binary incompatible with the same version without FT, at least it did at some point. Will check if requests are now padded to ensure ABI.

Reply:
The padding is only for predefined objects (aka. REQUEST_NULL). That being said from the user perspective all requests are pointers so what's behind that pointer should not really matter for binary compatibility (as long as it remains accesses only by the MPI library).

Reply:
All of this is gone. 

We now use pml_req->req_tag/peer when that's relevant, and we do not pretend to support request types that we anyway did not support.

We can deal with REQUEST_PML (and we will distinguish between COLL and P2P requests based on tags), REQUEST_COMM (the errors in the REQUEST_PML subreqs bubble up), and REQUEST_COLL (same story).